components/OlasTokenPage/SupplyPieChart.jsx

```diff
@@ -0,0 +1,200 @@
+import PropTypes from 'prop-types';
+import React, { useState, useEffect } from 'react';
+import Web3 from 'web3';
+import { Pie } from 'react-chartjs-2';
+import { Chart, ArcElement } from 'chart.js';
+import Verify from '../Verify';
+import olasAbi from '../../data/ABIs/Olas.json';
+
+// manually register arc element – required due to chart.js tree shaking
+Chart.register(ArcElement);
+
+const olasAddress = '0x0001A500A6B18995B03f44bb040A5fFc28E45CB0';
+const daoAddress = '0x3C1fF68f5aa342D296d4DEe4Bb1cACCA912D95fE';
+const buOlasAddress = '0xb09CcF0Dbf0C178806Aaee28956c74bd66d21f73';
+const veOlasAddress = '0x7e01A500805f8A52Fad229b3015AD130A332B7b3';
+const valoryAddress = '0x87cc0d34f6111c8A7A4Bdf758a9a715A3675f941';
+const providerUrl = 'https://ethereum.publicnode.com';
+
+const LABELS = [
+  'veOLAS (vote escrow)',
+  'DAO Treasury',
+  'buOLAS',
+  'Valory (core contributor)',
+  'Circulating supply',
+];
+
+const ADDRESSES = [
+  veOlasAddress,
+  daoAddress,
+  buOlasAddress,
+  valoryAddress,
+];
+const COLORS = {
+  'bg-purple-500': '#A755F7',
+  'bg-pink-500': '#E964C4',
+  'bg-orange-400': '#FFB246',
+  'bg-green-400': '#3FE681',
+  'bg-cyan-500 ': '#09B4D7',
+};
+
+const TAILWIND_COLOR = Object.keys(COLORS);
+const RGB_COLOR = Object.values(COLORS);
+
+function getAddressPrefix(address) {
+  return address.slice(0, 6);
+}
+
+const LegendItem = ({
+  label, color, address, value,
+}) => (
+  <div className="flex gap-2 items-center w-full">
+    <div className={`${color} px-3 py-1 rounded-sm`} />
+    <span className="text-gray-500">{label}</span>
+    {address && (
+      <span className="ml-1 font-medium">
+        <a
+          href={`https://etherscan.io/token/${olasAddress}?a=${address}`}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          {getAddressPrefix(address)}
+        </a>
+      </span>
+    )}
+    <span className="flex-auto border-b border-dotted border-gray-300" />
+    <span>{value}</span>
+  </div>
+);
+
+LegendItem.propTypes = {
+  label: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  address: PropTypes.string,
+  value: PropTypes.string.isRequired,
+};
+LegendItem.defaultProps = { address: null };
+
+function formatNumber(number) {
+  return number.toLocaleString();
+}
+
+function formatEthers(value) {
+  const weiValue = value;
+  const divisor = 1000000000000000000n;
+  const etherValue = weiValue / divisor;
+  return Number(etherValue);
+}
+
+const SupplyPieChart = () => {
+  const [data, setData] = useState([]);
+  const [totalSupply, setTotalSupply] = useState();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const web3 = new Web3(new Web3.providers.HttpProvider(providerUrl));
```

Comment:
I agree, that was annoying using this. But also think, let's maybe discuss that properly. It'd be a mess with all different approaches + we ideally need to switch to ts first to avoid all the problems we had with updating packages and etc.